src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,37 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/** Class representing a task which has a deadline */
+public class Deadline extends Task {
+    private LocalDate by;
+
+    /**
+     * Class constructor specifying the description of the deadline task.
+     * @param description the string description of the task.
+     * @param by the deadline of the task.
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = LocalDate.parse(by);
+    }
+
+    /**
+     * Returns a string description which contains the deadline of the task.
+     * @return a string representation of the deadline.
+     */
```

Comment:
For @return tags in Javadoc, it is actually indicated to be optional for members like toString() as they are [self-explanatory](https://google.github.io/styleguide/javaguide.html#s7.3.1-javadoc-exception-self-explanatory), so there is no problem in giving one-liners like this:

> /**
> &ensp;* Returns a string description which contains the deadline of the task.
> &ensp;*/

Same for your other files.